modelforge/tests/test_schnet.py

```diff
@@ -166,24 +168,24 @@ def test_compare_forward():
     # ---------------------------------------- #
     from modelforge.potential.schnet import SchNet
 
-    from modelforge.tests.test_models import load_configs
+    from modelforge.tests.test_models import load_configs_into_pydantic_models
 
     # load default parameters
-    config = load_configs(f"schnet", "qm9")
+    config = load_configs_into_pydantic_models(f"schnet", "qm9")
 
     # override default parameters
-    config["potential"]["core_parameter"]["featurization"][
-        "number_of_per_atom_features"
-    ] = 12
-    config["potential"]["core_parameter"]["number_of_radial_basis_functions"] = 5
-    config["potential"]["core_parameter"]["number_of_filters"] = 12
+    config["potential"].core_parameter.featurization.number_of_per_atom_features = 12
+    config["potential"].core_parameter.number_of_radial_basis_functions = 5
+    config["potential"].core_parameter.number_of_filters = 12
 
     torch.manual_seed(1234)
 
     # initialize model
     schnet = SchNet(
-        **config["potential"]["core_parameter"],
-        postprocessing_parameter=config["potential"]["postprocessing_parameter"],
+        **config["potential"].model_dump()["core_parameter"],
```

Comment:
slight preference towards `dict(dataclass)`

Reply:
dict(PydanticClass) does not recursively convert a nested model to a dictionary, only the top level.  In the case above, `dict(config['potential')['core_parameter']` would result in returning an instance of the CoreParameter pydantic class, not a dict of the contents. 

This is something I think we should actually discuss in terms of future road maps.  Do we want the functions to accept the pydantic models directly or maintain a pythonic interface of variables (or have a wrapper for the pythonic call that takes the pydantic model and does appropriate conversion).